### PR TITLE
Add DataTables extension colReorder to flowcells page

### PR DESCRIPTION
**Pull request by sofiahag**
Maybe we can start with one page, and try one that has data locally? I picked the flowcells page. You drag the header to the right or left to change the column order.

### CODE REVIEW

**Review by aanil — COMMENTED** [review]
Seems to break the formatting in the running notes column.

**Comment by sofiahag** [comment]
> Seems to break the formatting in the running notes column.

Huh that's what I saw this morning in production! Interesting, ok, something with the new running note generation? Do you know why this would be? 🤔 
There's no way to alter this code, so we would need to fix the running note code I guess 🤷🏻‍♀️

**Comment by aanil** [comment]
> > Seems to break the formatting in the running notes column.
> 
> Huh that's what I saw this morning in production! Interesting, ok, something with the new running note generation? Do you know why this would be? 🤔 There's no way to alter this code, so we would need to fix the running note code I guess 🤷🏻‍♀️

The javascript doesn't run after the column is moved. It wouldn't have run with the old version either.

**Comment by sofiahag** [comment]
> > > Seems to break the formatting in the running notes column.
> > 
> > 
> > Huh that's what I saw this morning in production! Interesting, ok, something with the new running note generation? Do you know why this would be? 🤔 There's no way to alter this code, so we would need to fix the running note code I guess 🤷🏻‍♀️
> 
> The javascript doesn't run after the column is moved. It wouldn't have run with the old version either.

You mean the code generating the new badges won't run? OK, guess we'll have to fix it some way if we want this extension 👍🏻

**Comment by aanil** [comment]
> > > > Seems to break the formatting in the running notes column.
> > > 
> > > 
> > > Huh that's what I saw this morning in production! Interesting, ok, something with the new running note generation? Do you know why this would be? 🤔 There's no way to alter this code, so we would need to fix the running note code I guess 🤷🏻‍♀️
> > 
> > 
> > The javascript doesn't run after the column is moved. It wouldn't have run with the old version either.
> 
> You mean the code generating the new badges won't run? OK, guess we'll have to fix it some way if we want this extension 👍🏻

No, the code generating the entire running note card does not run. See if you can get the move event to trigger the code generating the card.

**Comment by sofiahag** [comment]
> > > > > Seems to break the formatting in the running notes column.
> > > > 
> > > > 
> > > > Huh that's what I saw this morning in production! Interesting, ok, something with the new running note generation? Do you know why this would be? 🤔 There's no way to alter this code, so we would need to fix the running note code I guess 🤷🏻‍♀️
> > > 
> > > 
> > > The javascript doesn't run after the column is moved. It wouldn't have run with the old version either.
> > 
> > 
> > You mean the code generating the new badges won't run? OK, guess we'll have to fix it some way if we want this extension 👍🏻
> 
> No, the code generating the entire running note card does not run. See if you can get the move event to trigger the code generating the card.

Huh they generate for me 🤔 the markdown is broken though 🤷🏻‍♀️

**Comment by aanil** [comment]
> > > > > > Seems to break the formatting in the running notes column.
> > > > > 
> > > > > 
> > > > > Huh that's what I saw this morning in production! Interesting, ok, something with the new running note generation? Do you know why this would be? 🤔 There's no way to alter this code, so we would need to fix the running note code I guess 🤷🏻‍♀️
> > > > 
> > > > 
> > > > The javascript doesn't run after the column is moved. It wouldn't have run with the old version either.
> > > 
> > > 
> > > You mean the code generating the new badges won't run? OK, guess we'll have to fix it some way if we want this extension 👍🏻
> > 
> > 
> > No, the code generating the entire running note card does not run. See if you can get the move event to trigger the code generating the card.
> 
> Huh they generate for me 🤔 the markdown is broken though 🤷🏻‍♀️
> 
> Skärmavbild 2023-10-02 kl 10 43 18

I guess the card itself in the Flowcells queue is created in the html and the contents formatted in js.

**Comment by sofiahag** [comment]
> > > > > > > Seems to break the formatting in the running notes column.
> > > > > > 
> > > > > > 
> > > > > > Huh that's what I saw this morning in production! Interesting, ok, something with the new running note generation? Do you know why this would be? 🤔 There's no way to alter this code, so we would need to fix the running note code I guess 🤷🏻‍♀️
> > > > > 
> > > > > 
> > > > > The javascript doesn't run after the column is moved. It wouldn't have run with the old version either.
> > > > 
> > > > 
> > > > You mean the code generating the new badges won't run? OK, guess we'll have to fix it some way if we want this extension 👍🏻
> > > 
> > > 
> > > No, the code generating the entire running note card does not run. See if you can get the move event to trigger the code generating the card.
> > 
> > 
> > Huh they generate for me 🤔 the markdown is broken though 🤷🏻‍♀️
> > Skärmavbild 2023-10-02 kl 10 43 18
> 
> I guess the card itself in the Flowcells queue is created in the html and the contents formatted in js.

I can get the markdown to render for the card body with:

```
table.on( 'column-reorder', function() {
        $(".running-note-card > .card-body").each(function(i){
            $(this).html(make_markdown($(this).text()));
        });
});
```
    
   
But it isn't working for the card header, not sure why 🤔

**Comment by sofiahag** [comment]
Right, left out the badge code! Ok, it should work now! I can put it on stage, if no one is using it?

**Comment by sofiahag** [comment]
Ok, up on stage!

**Comment by sofiahag** [comment]
Wait, it's only working for some cards 😞 why????

File: run_dir/design/flowcells.html [hunk 1]
```diff
@@ -262,13 +262,26 @@ <h1><span id="illumina_page_title">Illumina flowcells</span>
...
         { extend: 'copy', className: 'btn btn-outline-dark mb-3' },
         { extend: 'excel', className: 'btn btn-outline-dark mb-3' }
       ]
     });
 
+    //Keep markdown format when dragging around the columns
+    table.on( 'column-reorder', function() {
+        $(".running-note-card > .card-body").each(function(i){
+            $(this).html(make_markdown($(this).text()));
+            $('.fillbadgecolour').html(function(){
```
Comment (aanil) [inline run_dir/design/flowcells.html:277]: This function `$('.fillbadgecolour').html(function(){` should be outside the `$(".running-note-card > .card-body").each(function(i){` function.

Comment (sofiahag) [reply run_dir/design/flowcells.html]: I have set sanity checks on myself, works now on stage 🤷🏻‍♀️

File: run_dir/design/flowcells.html [hunk 2]
```diff
@@ -262,13 +262,26 @@ <h1><span id="illumina_page_title">Illumina flowcells</span>
...
+    //Keep markdown format when dragging around the columns
+    table.on( 'column-reorder', function() {
+        $(".running-note-card > .card-body").each(function(i){
+            $(this).html(make_markdown($(this).text()));
+        });
+        $('.fillbadgecolour').html(function(){
+            let categories = JSON.parse($(this).text().replace(/'/g, '"'))
+            //from running_notes.js
+            return generate_category_label(categories)
+        })
```
Comment (aanil) [inline run_dir/design/flowcells.html:282]: Put these lines in a function, maybe called `format_running_note` and call it here and line 244? The code is repeated now.

Comment (sofiahag) [reply run_dir/design/flowcells.html:282]: Good idea!

Comment (sofiahag) [reply run_dir/design/flowcells.html]: On stage again :-)

**Review by aanil — APPROVED** [review]
👍 Looks good!

**Pull request merged**